File: app/src/androidTest/java/com/pavelnazimok/uitesting/java/robots/SecondScreenRobot.java
package com.pavelnazimok.uitesting.java.robots;

import com.pavelnazimok.uitesting.R;

import static com.pavelnazimok.uitesting.java.utils.UiAutomatorExtensions.findById;
import static com.pavelnazimok.uitesting.java.utils.UiAutomatorExtensions.waitUntilNotDisplayedById;
import static com.schibsted.spain.barista.interaction.BaristaClickInteractions.clickOn;

@SuppressWarnings("UnusedReturnValue")
public class SecondScreenRobot {

    public SecondScreenRobot clickAlertDialog() {
        clickOn(R.id.button_alert_dialog);
        return this;
    }

    public SecondScreenRobot clickOkInAlertDialog() {
        findById(android.R.id.button1, 10000).click();
        return this;
    }

    public SecondScreenRobot clickCancelInAlertDialog() {
        findById(android.R.id.button2, 10000).click();
        return this;
    }

    public SecondScreenRobot assertAlertDialogIsNotDisplayed() {
        waitUntilNotDisplayedById("alertTitle", 5000);
        return this;
    }
}

File: app/src/androidTest/java/com/pavelnazimok/uitesting/kotlin/robots/SecondScreenRobot.kt
package com.pavelnazimok.uitesting.kotlin.robots

import com.pavelnazimok.uitesting.R
import com.pavelnazimok.uitesting.kotlin.utils.findById
import com.pavelnazimok.uitesting.kotlin.utils.waitForView
import com.pavelnazimok.uitesting.kotlin.utils.waitUntilNotDisplayedById
import com.schibsted.spain.barista.interaction.BaristaClickInteractions.clickOn

fun onSecondScreen(func: SecondScreenRobot.() -> Unit): SecondScreenRobot {
    return SecondScreenRobot().apply {
        assertSecondScreenIsDisplayed()
        func()
    }
}

class SecondScreenRobot {

    fun clickAlertDialog() {
        clickOn(R.id.button_alert_dialog)
    }

    fun clickOkInAlertDialog() {
        findById(android.R.id.button1).click()
    }

    fun clickCancelInAlertDialog() {
        findById(android.R.id.button2).click()
    }

    fun assertSecondScreenIsDisplayed() {
        waitForView(R.id.screen_second)
    }

    fun assertAlertDialogIsNotDisplayed() {
        waitUntilNotDisplayedById("alertTitle")
    }
}
